### PR TITLE
36

### PR DESCRIPTION
**Pull request by CJ-Wright**
Rerender for py36

### CODE REVIEW

**Comment by jakirkham** [comment]
Why is the test getting removed?

**Comment by CJ-Wright** [comment]
@jakirkham It was giving an error (something about being the same dir and conda build 2.1+) and also it was removed in the other PR https://github.com/conda-forge/pymongo-feedstock/pull/1

File: recipe/run_test.sh [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-export DB_PATH="$SRC_DIR/temp-mongo-db"
-export LOG_PATH="$SRC_DIR/mongolog"
-export DB_PORT=27272
-export PID_FILE_PATH="$SRC_DIR/mongopidfile"
-
-mkdir "$DB_PATH"
-
-mongod --dbpath="$DB_PATH" --fork --logpath="$LOG_PATH" --port="$DB_PORT" --pidfilepath="$PID_FILE_PATH"
-
-pushd $SRC_DIR
```
Comment (jakirkham) [inline recipe/run_test.sh]: I think this line is the one causing problems with `conda-build` 2. We should look at using `source_files` in the `test` section instead.

Comment (CJ-Wright) [reply recipe/run_test.sh]: Is there a good way to copy the entire source dir?

Comment (jakirkham) [reply recipe/run_test.sh]: Probably adding `{{ SRC_DIR }}` or maybe even `.`. Though I hope we can do better than that. We should be able to just copy `tests` and then change this to `python -m unittest discover -s ./tests`.

Comment (CJ-Wright) [reply recipe/run_test.sh]: My hope is to get it working and then make the testing more elegant.

Comment (jakirkham) [reply recipe/run_test.sh]: That's fine. Wasn't encouraging this so much for elegance as avoiding more headache.

**Comment by CJ-Wright** [comment]
@jakirkham I'm not certain why this is broken on OSX it seems like a bunch of timeouts? Any suggestions? This may be a good reason to remove the `run_test.sh`.

**Comment by jakirkham** [comment]
Actually could try [this]( https://github.com/conda-forge/mosfit-feedstock/issues/23#issuecomment-349016207 ).

**Comment by jakirkham** [comment]
Not sure if we will need the same thing in the tests, but the CircleCI builds suggest we don't.

**Comment by CJ-Wright** [comment]
Admittedly I think it's worked on circle fairly consistently so?

**Comment by jakirkham** [comment]
Ah, but here's the thing. The CircleCI builds ran before `conda-build` 2.1.18 was release on `conda-forge`. So used `conda-build` 2.1.17 (can verify in the old logs). Hence why I think they didn't have the issue.

However the Travis CI builds ran significantly later and used `conda-build` 2.1.18 (again can verify in the logs). Hence they started running into connectivity issues. Thus the failures in the Travis CI builds appear related to this `conda-build` bug.

As further confirmation of this theory, please see issue ( https://github.com/conda-forge/mosfit-feedstock/issues/23 ) where both Linux and macOS builds failed due to connectivity issues. This has popped up on a few other feedstocks since `conda-build` 2.1.18 was packaged.

More details in this [comment]( https://github.com/conda-forge/mosfit-feedstock/issues/23#issuecomment-349016207 ).

**Comment by jakirkham** [comment]
Looked like we were seeing the same failure in the tests again. So went ahead and unset those environment variables for the tests in this PR. Let's see how it goes.

**Comment by CJ-Wright** [comment]
My colleague @chiahaoliu just ran this on his local OSX machine and got it to build. So this may be a travis related problem?

**Comment by jakirkham** [comment]
Was wondering that. Maybe Travis CI is starting another MongoDB daemon somewhere?

**Comment by CJ-Wright** [comment]
Would it be possible to let this to go through without osx and fix it later?

**Comment by CJ-Wright** [comment]
@jakirkham Can I please get some more review/merge?

File: recipe/run_test.sh [hunk 2]
```diff
@@ -1,17 +1,20 @@
 #!/bin/bash
+if [ "$(uname)" != "Darwin" ]; then
```
Comment (jakirkham) [inline recipe/run_test.sh]: So this isn't an issue when not on CI right? Maybe we should tweak this comparison to limit it to CI?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,7 +11,8 @@ source:
 
 build:
   number: 0
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  script: unset REQUESTS_CA_BUNDLE; unset SSL_CERT_FILE; python setup.py install --single-version-externally-managed --record=record.txt  # [not win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Sorry have forgotten. Did we need to unset these during the build too or only during testing?

File: recipe/run_test.sh (hunk 2 above)
Comment (jakirkham) [reply recipe/run_test.sh]: It wouldn't hurt to have a comment explaining that we encountered some issues contacting (maybe even starting?) the daemon process of MongoDB.

**Comment by jakirkham** [comment]
Thanks @CJ-Wright. Added a couple comments/questions.

I think the biggest thing we should do is have a comment (and maybe an issue on the feedstock) explaining the issue we had with Travis CI with a link to the log that demonstrates this. That way someone can pick it up again latter. The other things are not as important and can be saved for another PR (given Travis CI's sluggishness).

File: recipe/meta.yaml (hunk 3 above)
Comment (CJ-Wright) [reply recipe/meta.yaml]: I don't think we needed this at all on linux

File: recipe/run_test.sh [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 # We don't run on OSX the full tests due to timeout errors
-# See https://travis-ci.org/conda-forge/pymongo-feedstock/builds/311631125?utm_source=github_status&utm_medium=notification
+# See https://travis-ci.org/conda-forge/pymongo-feedstock/builds/311631125
```
Comment (CJ-Wright) [inline recipe/run_test.sh]: thank you!

Comment (jakirkham) [reply recipe/run_test.sh]: Of course.

**Comment by jakirkham** [comment]
LGTM. Thanks @CJ-Wright. Will merge when it passes.

**Comment by CJ-Wright** [comment]
Awesome! Thank you!

File: recipe/run_test.sh (hunk 2 above)
Comment (CJ-Wright) [reply recipe/run_test.sh]: What is the best way to check if this is running on travis?

**Comment by jakirkham** [comment]
Pushed some tweaks to get the Travis CI skip to stick. Planning on merging tomorrow if this looks ok.

**Pull request merged**